### PR TITLE
Peephole Optimize out conv(x).dim(), which prevents BN fusion

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #50241 Re-enable no-op peepholes
* #50222 Add Post Freezing Optimizations, turn on by default in torch.jit.freeze
* **#50221 Peephole Optimize out conv(x).dim(), which prevents BN fusion**
* #50220 [JIT] Factor out peephole to own test file
* #50075 [JIT] Add Frozen Conv-> Add/Sub/Mul/Div fusion
* #50074 [JIT] Frozen Graph Conv-BN fusion

Differential Revision: [D25856266](https://our.internmc.facebook.com/intern/diff/D25856266)

### CODE REVIEW

File: torch/csrc/jit/passes/peephole_alias_sensitive.cpp [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        // if we add more checks related to resize_ in the graph,
+        // factor this out like collectResizeSet in shape_analysis
+        if (!aliasDb_->hasWriters(node->output())) {
+          for (const Use& dim_use : dim_uses) {
+            replaceWithIValue(dim_use.user->output(), output_size);
+          }
+        } else {
+          for (const Use& dim_use : dim_uses) {
+            if (aliasDb_->moveAfterTopologicallyValid(node, dim_use.user)) {
+              replaceWithIValue(dim_use.user->output(), output_size);
```
Comment (Krovatkin) [inline torch/csrc/jit/passes/peephole_alias_sensitive.cpp:61]: Just to make sure I understand this part:
if we have multiple `dim`s that we can move `node` to (without any `resizes_` in between). We would move `conv` multiple times?

**Review by Krovatkin — APPROVED** [review]
:shipit:

File: torch/csrc/jit/passes/peephole_alias_sensitive.cpp (hunk 1 above)
Comment (eellison) [reply torch/csrc/jit/passes/peephole_alias_sensitive.cpp:61]: We're repeatedly trying to move `aten::dim` to the conv node. Not the other way around.

Comment (Krovatkin) [reply torch/csrc/jit/passes/peephole_alias_sensitive.cpp:61]: ah, misread the api comment, thanks!

**Pull request closed without merging**